creusot/src/backend/clone_map/elaborator.rs

```diff
@@ -330,25 +329,26 @@ impl DepElab for TyElab {
         match ty.kind() {
             TyKind::Alias(_, _) => vec![ctx.assoc_ty_decl(&mut names, def_id, subst)],
             _ => {
-                if let Some(why3_modl) = util::get_builtin(ctx.tcx, def_id) {
-                    let qname = QName::from_string(why3_modl.as_str());
-                    let Kind::Used(_, _) = names.insert(Dependency::Type(ty)) else {
+                if let Some(_why3_modl) = util::get_builtin(ctx.tcx, def_id) {
+                    let Kind::UsedBuiltin(name, _) = names.insert(Dependency::Type(ty)) else {
                         return vec![];
                     };
 
-                    let use_decl = Use { as_: None, name: qname.module_qname(), export: false };
+                    let use_decl = Use { as_: None, name: name.clone(), export: false };
 
                     vec![Decl::UseDecl(use_decl)]
                 } else {
-                    let name = names.insert(dep).qname();
+                    let k = names.insert(dep);
+                    let Kind::Used(modl, alias, _) = k else {
+                        debug!("TODO: expand: {:?}", k);
+                        return vec![];
```

Comment:
why is this correct?

Reply:
I'm actually not sure why. Unlike the above `todo!()` this branch is reached by the test suite and yet returning `vec![]` here doesn't change the output from before.

Reply:
I found that the vector returned in this case doesn't matter because it gets put in a map and is never looked up. I refactored this part again and left a comment about this.